app/controllers/public/loans_controller.rb

```diff
@@ -1,28 +1,34 @@
 class Public::LoansController < Public::PublicController
   include WordpressEmbeddable
+  after_action :verify_authorized
 
   def index
     params[:division] = get_division_from_url
-    @loans = Loan.filter_by_params(params).visible.
-      includes(:cooperative, division: :super_division).
-      paginate(:page => params[:pg], :per_page => 20).
-      order('signing_date DESC')
+    @params = { status: params[:status], pg: params[:pg], country: params[:country] }
+    @loans = policy_scope(Loan.filter_by_params(params).visible.
+          includes(:organization, division: :parent).
+          page(params[:pg]).per(20).
+          order('signing_date DESC'))
+    authorize @loans
     @countries = Country.order(:iso_code).pluck(:iso_code)
 
+
     # Set last loan list URL for 'Back to Loan List' link
     session[:loans_path] = request.fullpath
   end
 
   def show
-    @loan = Loan.status('all').find(params[:id])
+    @loan = Loan.find(params[:id])
     @pictures = @loan.featured_pictures(5) # for slideshow
-    @other_loans = @loan.cooperative.loans.status('all').order("SigningDate DESC") if @loan.cooperative
-    @repayments = @loan.repayments.order('DateDue')
+    @other_loans = policy_scope(Loan.related_loans(@loan))
+    
+    authorize @loan
   end
 
   def gallery
-    @loan = Loan.status('all').find(params[:id])
+    @loan = Loan.find(params[:id])
     @coop_media = @loan.coop_media(100, true).in_groups_of(4, false)
     @loan_media = (@loan.loan_media(100, true) + @loan.log_media(100, true)).in_groups_of(4, false)
+    authorize @loan
```

Comment:
Why put the authorize statements at the end?

Reply:
No particular reason, they could go anywhere after the @loan assignment, I can move them to be directly below it if you'd prefer.